app/views/admin/schedules/index.html.haml

```diff
@@ -25,8 +25,20 @@
                 method: :get, class: 'btn btn-primary'
                 = link_to 'Delete', admin_conference_schedule_path(@conference.short_title, schedule.id),
                 method: :delete, class: 'btn btn-danger', data: { confirm: "Do you really want to delete Schedule #{schedule.id}?" }
-
-.row
-  .col-md-12.text-right
-    = link_to 'Add Schedule', admin_conference_schedules_path(@conference.short_title),
-    method: :post, class: 'btn btn-primary'
+- if @rooms.nil?
+  .h3
+    No Venue!
+    %small
+      = link_to 'Create a venue with rooms', new_admin_conference_venue_path
+      before creating the schedule.
+- elsif @rooms.present?
+  .row
+    .col-md-12.text-right
+      = link_to 'Add Schedule', admin_conference_schedules_path(@conference.short_title),
+      method: :post, class: 'btn btn-primary'
+- else
+  .h3
+    No Rooms!
+    %small
+      = link_to 'Create rooms', admin_conference_venue_rooms_path
+      before creating the schedule.
```

Comment:
Please add an empty line at the end of the file. That's gonna make the red indication disappear. Your editor might have an option to do this automatically.